types/wlr_seat.c

```diff
@@ -67,33 +115,28 @@ static void wl_seat_destroy(struct wl_resource *resource) {
 		wl_resource_destroy(handle->pointer);
 	}
 	if (handle->keyboard) {
-		wl_resource_destroy(handle->pointer);
+		wl_resource_destroy(handle->keyboard);
 	}
 	if (handle->touch) {
-		wl_resource_destroy(handle->pointer);
+		wl_resource_destroy(handle->touch);
 	}
 	wl_signal_emit(&handle->wlr_seat->events.client_unbound, handle);
 	wl_list_remove(&handle->link);
 	free(handle);
 }
 
-static void wl_seat_release(struct wl_client *client,
-		struct wl_resource *resource) {
-	wl_seat_destroy(resource);
-}
-
 struct wl_seat_interface wl_seat_impl = {
 	.get_pointer = wl_seat_get_pointer,
 	.get_keyboard = wl_seat_get_keyboard,
 	.get_touch = wl_seat_get_touch,
-	.release = wl_seat_release
+	.release = resource_destroy
 };
 
 static void wl_seat_bind(struct wl_client *wl_client, void *_wlr_seat,
 		uint32_t version, uint32_t id) {
 	struct wlr_seat *wlr_seat = _wlr_seat;
 	assert(wl_client && wlr_seat);
-	if (version > 5) {
+	if (version > 6) {
```

Comment:
Was it intentional to only implement version 5? There is already version 6 of wl_seat (adds a pointer function or something iirc)